### PR TITLE
Switch from callback refs to createRef

### PR DESCRIPTION
**Pull request by akre54**
`createRef` is the preferred way of interacting with the DOM in React. This updates all callback refs to use the newer API.

### CODE REVIEW

File: src/components/map/layer-hover-info.js [hunk 1]
```diff
@@ -17,6 +17,7 @@
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
+
```
Comment (akre54) [inline src/components/map/layer-hover-info.js]: I also ran lint

**Comment by heshan0131** [comment]
@akre54 I rebased with master, build still seem to fail
https://deploy-preview-622--keplergl.netlify.com/demo

Can you make sure all the components you touched in this branch works?

**Comment by CharlieMartell** [comment]
I am seeing some issues related to the ref assigned to the typeahead component I believe this will fix `Uncaught TypeError: this.entry.focus is not a function`. Would be great to get this in :)

**Comment by akre54** [comment]
Sorry -- I made a change and missed that one.

The places changed in this PR are:
* File upload drag-n-drop
* File upload button
* Typeahead
* Range brush
* Range slider
* Slider
* Time slider
* Modal `rootNode`-- maybe we should use a [portal](https://reactjs.org/docs/portals.html) for this
* Map popover
* Plot container image capture -- currently only capturing map, no data overlay (related to #656 ?)

I went through these scenarios and they all seem fine with the exception of image capture, but would love another look.

**Comment by macrigiuseppe** [comment]
> I am seeing some issues related to the ref assigned to the typeahead component I believe this will fix `Uncaught TypeError: this.entry.focus is not a function`. Would be great to get this in :)

We found out the problem was related to styled-components outdate version on the consumer app

**Comment by akre54** [comment]
@macrigiuseppe @heshan0131 how are we looking for this?

**Comment by macrigiuseppe** [comment]
> @macrigiuseppe @heshan0131 how are we looking for this?

Hey @akre54, can you rebase with the current version of master. @heshan0131 just fixed the export image feature and you will be able to test it the export image flow as well

File: src/components/common/time-slider-marker.js [hunk 2]
```diff
@@ -115,11 +113,8 @@ export default class TimeSliderMarker extends Component {
         className="time-slider-marker"
         width={this.props.width}
         height={height}
-        ref={comp => {
-          this.svgContainer = comp;
-        }}
       >
-        <g className="x axis" transform="translate(0, 0)" />
+        <g ref={this.xAxis} transform="translate(0, 0)" />
```
Comment (heshan0131) [inline src/components/common/time-slider-marker.js]: Need the className here for style

**Comment by heshan0131** [comment]
Seeing some error here:

1. File upload drag-n-drop
When drag-n-drop a file into kepler, like Add Data button again. File drag-n-drop disappear
Screen Shot 2019-08-09 at 1 48 11 PM

2. Time slider axis color should be white, not is black, missing className
Screen Shot 2019-08-13 at 12 24 04 PM

**Comment by akre54** [comment]
Fixed! Thanks for pointing out

**Comment by heshan0131** [comment]
Looks Good. Please commit with sign off to pass DCO

**Pull request merged**